app/mixins/fetch_package_detail.js

```diff
@@ -0,0 +1,28 @@
+import Ember from "ember";
+import _ from "lodash";
+
+export default Ember.Mixin.create({
+  fetchPackageDetails: Ember.computed("packageDetails", function() {
```

Comment:
This is a mixin, but it doesn't look very re-usable.
It makes the assumption that the class I included it in has the following properties `packageDetails, showAdditionalFields insertFixedOption`. 

If it's just a mean of extracting code outside of a controller then I'd suggest helper classes.

Maybe something like `utils/subform_details.js`

```javascript
// utils/subform_details.js
export function buildSubformModel(c, ...) {
   ...
}

// controller
{
    fetchPackageDetails: Ember.computed("packageDetails", function() {
         helper.buildSubformModel( ... )
    }
}
```

If it actually is meant to be used in multiple places, then I suggest documenting how to integrate it with a comment. Basically setting default values to  the required properties and explaining in a comment that this mixin works by reading X,Y,Z props

Side note: The name `fetchPackageDetails` also sounds like a function, bit confusing, how about simply `subFormData` ?

Reply:
it is not possible to move it in a Utils folder as its not a ember class and services are not accessible over there